app/scripts/modules/amazon/src/instance/awsInstanceType.service.js

```diff
@@ -297,12 +298,44 @@ module(AMAZON_INSTANCE_AWSINSTANCETYPE_SERVICE, []).factory('awsInstanceTypeServ
       return families.ebsOptimized.includes(family);
     }
 
+    function isBurstingSupported(instanceType) {
+      if (!instanceType) {
+        return false;
+      }
+      const [family] = instanceType.split('.');
+      return families.burstablePerf.includes(family);
+    }
+
+    function isInstanceTypeInCategory(instanceType, category) {
+      if (!instanceType || !category) {
+        return false;
+      }
+
+      if (category === 'custom') {
+        return true;
+      }
+
+      const [family] = instanceType.split('.');
+      let familyInCategory = defaultCategories
+        .filter((c) => c.type === category)[0]
+        .families.filter((t) => t.type === family);
+
+      let isTypeInCategory =
+        familyInCategory &&
+        familyInCategory.length > 0 &&
+        familyInCategory[0].instanceTypes.map((t) => t.name).includes(instanceType);
+
+      return isTypeInCategory;
```

Comment:
We can use lodash `find` (https://lodash.com/docs/4.17.15#find) here to clean this up. Also, use `const` instead of `let` since there is no reassignment happening. 

```js
const instanceCategory = find(defaultCategories, c => c.type === category);
const familyInCategory = find(instanceCategory?.families, t => t.type === family);

return familyInCategory && familyInCategory.instanceTypes.map((t) => t.name).includes(instanceType);
```